beacon-chain/rpc/beaconv1/state.go

```diff
@@ -174,12 +174,18 @@ func (bs *Server) state(ctx context.Context, stateId []byte) (*statetrie.BeaconS
 		}
 	case "finalized":
 		checkpoint := bs.ChainInfoFetcher.FinalizedCheckpt()
+		if checkpoint == nil {
+			return nil, status.Errorf(codes.NotFound, "Finality not yet achieved")
+		}
 		s, err = bs.StateGenService.StateByRoot(ctx, bytesutil.ToBytes32(checkpoint.Root))
 		if err != nil {
 			return nil, status.Errorf(codes.Internal, "Could not get finalized state: %v", err)
 		}
 	case "justified":
 		checkpoint := bs.ChainInfoFetcher.CurrentJustifiedCheckpt()
+		if checkpoint == nil {
```

Comment:
Same here, it isn't possible to retrieve a nil checkpoint
https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/blockchain/chain_info.go#L81